packages/frontend-core/src/components/ClientAppSkeleton.svelte

```diff
@@ -0,0 +1,346 @@
+<script>
+  export let sideNav = false
+  export let hideDevTools = false
+  export let hideFooter = false
+  export let noAnimation = false
+</script>
+
+<svelte:head>
+  <style>
+    @media (max-width: 720px) {
+      #clientAppSkeletonLoader .devToolsButton {
+        display: none;
+      }
+
+      #clientAppSkeletonLoader .main {
+        flex-direction: column;
+        width: initial;
+      }
+
+      #clientAppSkeletonLoader .nav {
+        height: 61px;
+        width: initial;
+      }
+
+      #clientAppSkeletonLoader .body {
+        width: initial;
+        height: 100%;
+      }
+
+      #clientAppSkeletonLoader .footer {
+        border-top: none;
+      }
+    }
+
+    @container (max-width: 720px) {
+      #clientAppSkeletonLoader .devToolsButton {
+        display: none;
+      }
+
+      #clientAppSkeletonLoader .main {
+        flex-direction: column;
+        width: initial;
+      }
+
+      #clientAppSkeletonLoader .nav {
+        height: 61px;
+        width: initial;
+      }
+
+      #clientAppSkeletonLoader .body {
+        width: initial;
+        height: 100%;
+      }
+
+      #clientAppSkeletonLoader .footer {
+        border-top: none;
+      }
+    }
+  </style>
+</svelte:head>
+
+<div class:sideNav id="clientAppSkeletonLoader" class="skeleton">
+  <div class="animation" class:noAnimation />
+
+  {#if !hideDevTools}
+    <div class="devTools">
```

Comment:
This dev tools implementation is potentially a little fragile to change, so i'd appreciate thoughts on it.

Importing the actual dev tools into this component to share some of the styles is a potential option, but the way we render svelte components on the server is very bare bones, and it doesn't handle imports at all, so you need to manually compile each component, or have your build step copy them in, so I opted to do it this way.

Reply:
For dev tools, could we just have the black background and nothing else? I think that perfectly fits in line with a skeleton. It means the implementation here can be extremely simple since its just a single empty div, but it still means that the page layout is correct. It would just look like the content loads in when it's ready, which I think would be perfectly fine.

I think that's definitely a worthy tradeoff to keep this implementation simpler and less fragile.